File: BackEnd/DotNet/WebDigital/WebDigital/dto/ProductCreateUpdateDto.cs
namespace WebDigital.dto
{
    public class ProductCreateUpdateDto
    {
        public string ProductName { get; set; } = null!;

        public string ProductIsbn { get; set; } = null!;

        public string? ProductDescriptionShort { get; set; }

        public string? ProductDescriptionLong { get; set; }

        public string? ProductImage { get; set; }

        public decimal ProductBaseprice { get; set; }

        public decimal? ProductOfferprice { get; set; }

        public DateOnly? ProductOffPriceExpirydate { get; set; }

        public decimal? DiscountPercent { get; set; }

        public decimal? RentPerDay { get; set; }

        public int? MinRentDays { get; set; }

        public bool IsRentable { get; set; }

        public bool IsLibrary { get; set; }

        public decimal? RoyaltyPercent { get; set; }

        public int? AttributeId { get; set; }

        public int? ProductAuthor { get; set; }

        public int? ProductGenere { get; set; }

        public int? ProductLang { get; set; }

        public int? ProductType { get; set; }

        public int? ProductPublisher { get; set; }
    }
}

File: BackEnd/Java/product-service/src/main/java/com/digitalshelf/productservice/dto/ProductCreateUpdateDto.java
package com.digitalshelf.productservice.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
public class ProductCreateUpdateDto {
    private String productName;
    private String productIsbn;
    private String productDescriptionShort;
    private String productDescriptionLong;
    private String productImage;
    private BigDecimal productBaseprice;
    private BigDecimal productOfferprice;
    private LocalDate productOffPriceExpirydate;
    private BigDecimal discountPercent;
    private BigDecimal rentPerDay;
    private Integer minRentDays;
    private boolean isRentable;
    private boolean isLibrary;
    private BigDecimal royaltyPercent;
    private Integer attributeId;
    private Integer productAuthor;
    private Integer productGenere;
    private Integer productLang;
    private Integer productType;
    private Integer productPublisher;
}